tests/e2e/upgrade_test.go

```diff
@@ -95,6 +96,34 @@ var _ = Describe("Upgrade", Label(tests.LabelUpgrade, tests.LabelNoOpenshift), O
 		if testLevelEnv.Depth < int(level) {
 			Skip("Test depth is lower than the amount requested for this test")
 		}
+
+		err := env.DeleteNamespace(operatorNamespace)
+		if !apierrs.IsNotFound(err) {
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		err = env.CreateNamespace(operatorNamespace)
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
This is going to happen on top of the `cleanupNamespace` and `assertCreateNamespace` we already have.
Locally I'm seeing problems with the operator namespace being there or not in unexpected ways.
Why is this necessary?

Reply:
As the comment https://github.com/cloudnative-pg/cloudnative-pg/pull/3624#issuecomment-1875083582 said, creating a namespace before creating a pull image secret